src/pykeen/datasets/ea/combination.py

```diff
@@ -44,69 +292,208 @@ def __call__(
+            dict(entity_offsets=offsets[:, 0], relation_offsets=offsets[:, 1]),
+        )
+
+
 class ExtraRelationGraphPairCombinator(GraphPairCombinator):
     """This combinator keeps all entities, but introduces a novel alignment relation."""
 
-    def __call__(
+    #: the name of the additional alignment relation
+    ALIGNMENT_RELATION_NAME: ClassVar[str] = "same-as"
+
+    # docstr-coverage: inherited
+    def process(
         self,
-        left: TriplesFactory,
-        right: TriplesFactory,
-        alignment: pandas.DataFrame,
-        **kwargs,
-    ) -> Tuple[TriplesFactory, torch.LongTensor]:  # noqa: D102
-        mapped_triples = []
-        entity_to_id: Dict[str, int] = {}
-        relation_to_id: Dict[str, int] = {}
-        entity_offset = relation_offset = 0
-        entity_offsets = []
-        for side, tf in ((EA_SIDE_LEFT, left), (EA_SIDE_RIGHT, right)):
-            mapped_triples.append(
-                tf.mapped_triples + torch.as_tensor(data=[entity_offset, relation_offset, entity_offset]).view(1, 3)
-            )
-            entity_to_id.update((f"{side}:{key}", value + entity_offset) for key, value in tf.entity_to_id.items())
-            relation_to_id.update(
-                (f"{side}:{key}", value + relation_offset) for key, value in tf.relation_to_id.items()
-            )
-            entity_offsets.append(entity_offset)
-            entity_offset += tf.num_entities
-            relation_offset += tf.num_relations
-
-        # extra alignment relation
-        relation_to_id["same-as"] = relation_offset
-        # filter alignment
-        mask = ~(alignment[EA_SIDE_LEFT].isin(left.entity_to_id) & alignment[EA_SIDE_RIGHT].isin(right.entity_to_id))
-        if mask.any():
-            alignment = alignment.loc[~mask]
-            logger.warning(
-                f"Dropped {format_relative_comparison(part=mask.sum(), total=alignment.shape[0])} "
-                f"alignments due to unknown labels.",
-            )
-        # map alignment to (new) IDs
-        left_id = (
-            alignment[EA_SIDE_LEFT].apply(left.entity_to_id.__getitem__) + entity_offsets[0]
-        )  # offset should be zero
-        right_id = alignment[EA_SIDE_RIGHT].apply(right.entity_to_id.__getitem__) + entity_offsets[1]
-        # append alignment triples
-        mapped_triples.append(
-            torch.stack(
-                [
-                    torch.as_tensor(data=left_id.values, dtype=torch.long),
-                    torch.full(size=(len(left_id),), fill_value=relation_offset),
-                    torch.as_tensor(data=right_id.values, dtype=torch.long),
-                ],
-                dim=-1,
-            )
+        mapped_triples: MappedTriples,
+        alignment: torch.LongTensor,
+        offsets: torch.LongTensor,
+    ) -> ProcessedTuple:  # noqa: D102
+        # add alignment triples with extra relation
+        left_id, right_id = alignment
+        alignment_relation_id = offsets[-1, 1]
```

Comment:
This leads `alignment_relation_id` being a tensor and the since the relation offset is equal to the number of relations of the first graph the value `alignment_relation_id` is equal to the id of the first relation of the second graph

Reply:
cf. 7105ebb